.github/workflows/release.yml

```diff
@@ -0,0 +1,121 @@
+jobs:
+  build-distribution:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Build source and wheel distributions
+        run: |
+          python -m pip install --upgrade build twine
+          python -m build
+          twine check --strict dist/*
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  publish-test-pypi:
+    name: Upload release to PyPI Test Server
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v20')
+    needs:
+      - build-distribution
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://test.pypi.org/p/catalystcoop.pudl
+    permissions:
+      id-token: write
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+      - name: Publish distribution to PyPI Test Server
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+
+  #publish-pypi:
+  #  name: Upload release to PyPI
+  #  if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v20')
+  #  needs:
+  #    - build-distribution
+  #  runs-on: ubuntu-latest
+  #  environment:
+  #    name: pypi
+  #    url: https://pypi.org/p/catalystcoop.pudl
+  #  permissions:
+  #    id-token: write
+  #  steps:
+  #    - name: Download all the dists
+  #      uses: actions/download-artifact@v3
+  #      with:
+  #        name: python-package-distributions
+  #        path: dist/
+  #    - name: Publish distribution to PyPI
+  #      uses: pypa/gh-action-pypi-publish@release/v1
+
+  publish-github:
+    name: Release package on GitHub
+    needs:
+      - publish-test-pypi
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+      - name: Sign dist with Sigstore
+        uses: sigstore/gh-action-sigstore-python@v2.1.0
+        with:
+          inputs: ./dist/*.tar.gz ./dist/*.whl
+      - name: Upload artifact signatures to GitHub Release
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        run: >-
+          gh release create '${{ github.ref_name }}' ./dist/*.tar.gz ./dist/*.whl
+          --generate-notes
+          --repo '${{ github.repository }}'
+
+  notify-slack:
+    runs-on: ubuntu-latest
+    needs:
+      - publish-github
+    if: ${{ always() }}
+    steps:
+      - name: Inform the Codemonkeys
+        uses: 8398a7/action-slack@v3
```

Comment:
non-blocking: I think we can use the official slack-github-action here, so long as we set `SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK` in the env as well.

Reply:
I have no idea how any of the slack notifications work. I'm just cutting and pasting from other jobs...